woof-code/rootfs-skeleton/usr/sbin/xkbconfigmanager

```diff
@@ -492,53 +529,60 @@ NOTE: if you select a variant for a layout that already has one,
+		# adding var for gettext
+		export gtxt1="$(gettext 'Select XkbModel')"
+		# create dialog with all options (in temp file)
+		echo -n '#!/bin/sh
 Xdialog --left --stdout --title "$gtxt1" --menubox "' >$TMPDIALOG
-	echo -n "$MESSAGE\" 0 0 15 " >>$TMPDIALOG
-	cat $TMPLIST | tr '\n' ' ' >>$TMPDIALOG
-	chmod 755 $TMPDIALOG
-	# run dialog and get choice
-	CHOICE=`eval $TMPDIALOG`
-	[ "$CHOICE" = "" ] && echo "cancelled" && exit
-	echo "$CHOICE"
-	# find the variant chosen and its position
-	NEWVAR=`fgrep "$CHOICE" $TMPLIST | cut -d'"' -f2 | cut -d' ' -f2`
-	ITSLAY=`fgrep "$CHOICE" $TMPLIST | cut -d'"' -f4 | cut -d: -f1`
-	ITSPOS=`echo "$LAYOUTS" | tr ',' '\n' | fgrep -n "$ITSLAY" | cut -d: -f1`
-	#the last sed is since we need to cut off the last , (echo -n doesn't work well)
-	NEW=`echo "$OLD" | tr ',' '\n' | sed "${ITSPOS}s/.*/$NEWVAR/" | tr '\n' ',' | sed 's/,$//'`
-	# if old is same as new: remove
-	[ "$NEW" = "$OLD" ] && NEW=`echo "$OLD" | tr ',' '\n' | sed "${ITSPOS}s/.*//" | tr '\n' ',' | sed 's/,$//'`
-	if [ -f $XORG_CONF ]; then
-		# add new variant to line, or entire line, if it doesn't exist:
-		if [ "$ORIGINAL" = "" ] ;then
-			ORIGINAL="$LAYOUTSLINE"
-			MODIFIED="$ORIGINAL\n	Option      \"XkbVariant\" \"$NEW\""
-		else
-			MODIFIED=`echo "$ORIGINAL" | sed "s%\"$CURRENT\"%\"$NEW\"%"`
+		echo -n "$MESSAGE\" 0 0 15 " >>$TMPDIALOG
+		cat $TMPLIST | tr '\n' ' ' >>$TMPDIALOG
+		chmod 755 $TMPDIALOG
+		# run dialog and get choice
+		CHOICE=`eval $TMPDIALOG`
+		[ "$CHOICE" = "" ] && echo "cancelled" && exit
+		echo "$CHOICE"
+		# find the variant chosen and its position
+		NEWVAR=`fgrep "$CHOICE" $TMPLIST | cut -d'"' -f2 | cut -d' ' -f2`
+		ITSLAY=`fgrep "$CHOICE" $TMPLIST | cut -d'"' -f4 | cut -d: -f1`
+		ITSPOS=`echo "$LAYOUTS" | tr ',' '\n' | fgrep -n "$ITSLAY" | cut -d: -f1`
+		#the last sed is since we need to cut off the last , (echo -n doesn't work well)
+		NEW=`echo "$OLD" | tr ',' '\n' | sed "${ITSPOS}s/.*/$NEWVAR/" | tr '\n' ',' | sed 's/,$//'`
+		# if old is same as new: remove
+		[ "$NEW" = "$OLD" ] && NEW=`echo "$OLD" | tr ',' '\n' | sed "${ITSPOS}s/.*//" | tr '\n' ',' | sed 's/,$//'`
+		if [ -f $XORG_CONF ]; then
+			# add new variant to line, or entire line, if it doesn't exist:
+			if [ "$ORIGINAL" = "" ] ;then
+				ORIGINAL="$LAYOUTSLINE"
+				MODIFIED="$ORIGINAL\n	Option      \"XkbVariant\" \"$NEW\""
+			else
+				MODIFIED=`echo "$ORIGINAL" | sed "s%\"$CURRENT\"%\"$NEW\"%"`
+			fi
+			# add to file
+			sed -i "s%$ORIGINAL%$MODIFIED%" $XORG_CONF
+			if grep -q '"XkbModel"' $XORG_CONF ; then
+				XkbApplyNow
+			else
+				eval $ERRMSG
+			fi
 		fi
-		# add to file
-		sed -i "s%$ORIGINAL%$MODIFIED%" $XORG_CONF
-		if grep -q '"XkbModel"' $XORG_CONF ; then
-			XkbApplyNow
-		else
-			eval $ERRMSG
+	
+		if [ -f $XWAYLANDRC ]; then
+			sed -i "s%^XKB_DEFAULT_VARIANT=.*%XKB_DEFAULT_VARIANT=$NEW%" $XWAYLANDRC
+			[ -n "$WAYLAND_DISPLAY" ] && XkbApplyLater
```

Comment:
XkbApplyLater should be called only once. I don't think the use case of having both labwc and Cage is useful, so it's OK not to fix this.

Reply:
On condition it's called either if $XWAYLANDRC or $LABWCRC exists, and as you allude to, it's unlikely that cage and labwc will coexist.